### PR TITLE
[INLONG-3352][DataProxy] Fix DataProxy keeps trying to send messages that were sent failed

### LINKED ISSUE #3352: [Bug][Dataproxy]  Dataproxy keeps trying to send messages that have send failed
### What happened

Dataproxy keeps trying to send messages that have send failed

### What you expected to happen

Messages that fail to send should be resent a limited number of times

### How to reproduce

send order message use one topic is not exist.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
Fixed bug: DataProxy keeps trying to send messages that have send failed 

Fixes #3352

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java [hunk 1]
```diff
@@ -723,6 +727,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             SocketAddress remoteSocketAddress = remoteChannel.remoteAddress();
             if (!checkMessageTopic || !MessageUtils.isSyncSendForOrder(commonAttrMap
                     .get(AttributeConstants.MESSAGE_SYNC_SEND))) {
+                logger.debug("response 11111!");
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java:730]: comment of lacking meaning

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventStat.java [hunk 2]
```diff
@@ -17,20 +17,29 @@
...
 import org.apache.flume.Event;
+import org.apache.inlong.dataproxy.base.OrderEvent;
+import org.apache.inlong.dataproxy.utils.MessageUtils;
 
 public class EventStat {
+
+    private static long RETRY_INTERVAL_MS = 1 * 1000L;
     private Event event;
-    private int myRetryCnt;
+    private AtomicInteger myRetryCnt = new AtomicInteger(0);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventStat.java:29]: It is better to retry count use CountDownLatch.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/ServerMessageHandler.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventStat.java (hunk 2 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/EventStat.java:29]: The retry mechanism will be redesigned later

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**